flink-python/src/main/java/org/apache/flink/client/python/PythonEnvUtils.java

```diff
@@ -93,65 +91,60 @@ public void run() {
 	 * Prepares PythonEnvironment to start python process.
 	 *
 	 * @param pythonLibFiles The dependent Python files.
+	 * @param tmpDir The temporary directory files copied to.
 	 * @return PythonEnvironment the Python environment which will be executed in Python process.
 	 */
-	public static PythonEnvironment preparePythonEnvironment(List<Path> pythonLibFiles) throws IOException {
+	public static PythonEnvironment preparePythonEnvironment(
+				List<Path> pythonLibFiles,
+				String tmpDir) throws IOException {
 		PythonEnvironment env = new PythonEnvironment();
 
-		// 1. setup temporary local directory for the user files
-		String tmpDir = System.getProperty("java.io.tmpdir") +
-			File.separator + "pyflink" + File.separator + UUID.randomUUID();
+		tmpDir = new File(tmpDir).getAbsolutePath();
 
+		// 1. setup temporary local directory for the user files
 		Path tmpDirPath = new Path(tmpDir);
 		FileSystem fs = tmpDirPath.getFileSystem();
-		if (fs.exists(tmpDirPath)) {
```

Comment:
It is unnecessary because the directory passed to this method is always empty. In addition we should not delete the files which is not created by ourselves.